app/src/main/resources/static/js/posts.js

```diff
@@ -0,0 +1,12 @@
+var moreNum = 5;
+
+window.onload = function() {
+    $('.areaPost:nth-child(n + ' + (moreNum + 1) + ')').addClass('is_hidden');
+}
+
+$(document).on('click', '.buttonMore button', function() {
```

Comment:
```suggestion
$('.buttonMore button').on('click', function() {
```
みたいな感じでいけませんか？
さらに言えばボタン要素にid付与した方が良いかと思います。

jQueryに慣れているのでしょうか？
慣れているから時間短縮のためなら良いですが、jQueryは割と衰退してるので、ECMAでできるならそっちで済ませる方が良いかと思いました
https://www.w3schools.com/jsref/event_onclick.asp

Reply:
私も最初それで試していたのですが、そちらではクリックに反応しなかったため(書き方が悪いのか)今の記述にしました...
id付与は確かにそうですね。
jQueryに慣れているというわけではないので、そちらも検討してみます。ありがとうございます！